db/migrate/20170602111120_add_timestamps_to_documents.rb

```diff
@@ -0,0 +1,108 @@
+class AddTimestampsToDocuments < ActiveRecord::Migration[5.1]
+  def change
+    add_timestamps :documents, null: true
+    puts "> Timestamp fields added."
+    puts "> Updating Non-EN documents without editions..."
+
+    update_non_en_documents_without_editions
+    puts "> Non-EN documents without editions updated."
+    puts "> Updating remaining documents..."
+
+    Document.where(created_at: nil).find_each do |document|
+      editions = Edition.where(document_id: document.id).order(:user_facing_version)
+      if editions.present?
+        update_timestamps_from_editions(document, editions)
+      else
+        update_timestamps_from_events(document)
+      end
+    end
+
+    puts "> Remaining documents updated."
+    puts "> Setting NOT NULL constraint on created_at and updated_at..."
+    change_column_null :documents, :created_at, false
+    change_column_null :documents, :updated_at, false
+  end
+
+private
+
+  NON_EN_DOCUMENTS_WITHOUT_EDITIONS = [
+    {
+      content_id: "f091d705-8221-40c4-bc5d-2914c2d0763f",
+      locale: "en",
```

Comment:
Ah yes, my mistake this should be documents with multiple locales, not just non-EN.  Will update the naming 👍